src/httpd.c

```diff
@@ -1485,3 +1491,39 @@ httpd_deinit(void)
   evhttp_free(evhttpd);
   event_base_free(evbase_httpd);
 }
+
+void
+httpd_send_error(struct evhttp_request* req, int error, const char* reason)
```

Comment:
Please add explanation and reference to the original libevent source. Also: The function is misplaced here at the end of the file, it should of course come after httpd_send_reply.